contrib/k6/baggage.go

```diff
@@ -0,0 +1,67 @@
+package k6
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/grafana/pyroscope-go"
+	"go.opentelemetry.io/otel/baggage"
+)
+
+// LabelsFromBaggageHandler is a middleware that will extract key-value pairs
+// from the request baggage and make them profiling labels.
+func LabelsFromBaggageHandler(handler http.Handler) http.Handler {
+	lh := &labelHandler{
+		innerHandler: handler,
+	}
+
+	return lh
+}
+
+type labelHandler struct {
+	innerHandler http.Handler
+}
+
+func (lh *labelHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	labels := getBaggageLabels(r)
+	if labels == nil {
+		lh.innerHandler.ServeHTTP(w, r)
+		return
+	}
+
+	pyroscope.TagWrapper(r.Context(), *labels, func(ctx context.Context) {
+		lh.innerHandler.ServeHTTP(w, r.WithContext(ctx))
+	})
```

Comment:
What do you think about:

> Consider using [pprof.SetGorouteLabels](https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/runtime/pprof/runtime.go;l=47-52) instead: TagWrapper and pprof.Do introduce some overhead and also cause appearance of quite noisy frames in the stack traces

Reply:
I agree with this. This slipped through the cracks during my initial pass addressing your suggestions.

Reply:
Fixed in https://github.com/grafana/pyroscope-go/pull/99/commits/539601a7ef9ef8fa03e36bd22ee6d05e1d070aa3.